src/enzo/Grid_SchrodingerSolver.C

```diff
@@ -0,0 +1,350 @@
+    zpos = CellLeftEdge[2][k] + 0.5*CellWidth[2][k] -
+      PointSourceGravityPosition[2];
+    
+  for (j = 0; j < GridDimension[1]; j++) {
+    if (GridRank > 1)
+      ypos = CellLeftEdge[1][j] + 0.5*CellWidth[1][j] -
+        PointSourceGravityPosition[1];
+      
+    for (i = 0; i < GridDimension[0]; i++, n++) {
+      xpos = CellLeftEdge[0][i] + 0.5*CellWidth[0][i] -
+        PointSourceGravityPosition[0];
+
+      /* Compute distance from center. */
+ 
+      rsquared = xpos*xpos + ypos*ypos + zpos*zpos;
+      radius = sqrt(rsquared);
+      rcore = PointSourceGravityCoreRadius;
+      /* Compute potential from point source */
+
+      potential = -min(PointSourceGravityConstant/radius, PointSourceGravityConstant/rcore);
+        
+      auxre = cos(potential / hmcoef * dtFixed ) * repsi[n] 
+              + sin(potential / hmcoef * dtFixed ) * impsi[n];
+
+      auxim = cos(potential / hmcoef * dtFixed ) * impsi[n] 
+              - sin(potential / hmcoef * dtFixed ) * repsi[n];
+
+      repsi[n] = auxre;
+      impsi[n] = auxim;
+
+    } // end: loop over i
+  } // end: loop over j
+      } // end: loop over k
+  } // end: if point source
+
+  // Runge-Kutta 4th order
+  if ( rk4(repsi, impsi, 
+          GridDimension[0], GridDimension[1], GridDimension[2], GridRank, 
+          dtFixed, CellWidthTemp[0], CellWidthTemp[1], CellWidthTemp[2], hmcoef) == FAIL){
+    fprintf(stderr, "P(%"ISYM"): Error in rk4 on step %"ISYM" (dt=%"GSYM")\n", MyProcessorNumber,
+      nhy, dtFixed);
+    fprintf(stderr, "  grid dims = %"ISYM" %"ISYM" %"ISYM"\n", GridDimension[0], GridDimension[1], GridDimension[2]);
+    ENZO_FAIL("Error in 4th runge kutta advance\n");
+    }
+
+  // Stormer-Verlet 2nd order
+  /*if ( sv2(nhy,repsi, impsi, 
+          GridDimension[0], GridDimension[1], GridDimension[2], GridRank, 
+          dtFixed, CellWidthTemp[0], CellWidthTemp[1], CellWidthTemp[2], hmcoef) == FAIL){
+    fprintf(stderr, "P(%"ISYM"): Error in rk4 on step %"ISYM" (dt=%"GSYM")\n", MyProcessorNumber,
+      nhy, dtFixed);
+    fprintf(stderr, "  grid dims = %"ISYM" %"ISYM" %"ISYM"\n", GridDimension[0], GridDimension[1], GridDimension[2]);
+    ENZO_FAIL("Error in 4th runge kutta advance\n");
+    }*/
+
+
+  // Add half gravity after the TVD-RK
+  /*if (GravityOn && (PotentialField != NULL) ){
+    if(SchrdingerAddPotential(repsi, impsi,
+         GridDimension[0], GridDimension[1], GridDimension[2],
+         GridRank, 
+         GravitatingMassFieldDimension[0],GravitatingMassFieldDimension[1],GravitatingMassFieldDimension[2],
+         is, ie, js, je, ks, ke, 
+         dtFixed/2., hmcoef,
+         PotentialField, 
+         Offset[0], Offset[1], Offset[2]) == FAIL){
+         ENZO_FAIL("Error in Add poteitial after TVD-RK!\n");
+    }
+  }*/
+
+
+  // Update New Density
+
+  for (int i=0; i<size; i++){
+    d[i] = repsi[i]*repsi[i]+impsi[i]*impsi[i];
+    /*
+    // add a few dissipation, dens acts as absorbing layer, only work for FDMCollapse !!
```

Comment:
Should this commented section be switched to a flag checking if the simulation problem type is FDMCollapse vs. Cosmology?